MovieApp/MovieApp/Views/Auth/ViewModel/AuthViewModel.swift

```diff
@@ -18,44 +19,59 @@ class AuthViewModel: AuthViewModelProtocol, ObservableObject {
     
     // MARK: - Init
 
-    init(authService: AuthServiceProtocol, keychainService: KeychainServiceProtocol, coordinator: AuthCoordinatorProtocol) {
+    init(authService: AuthServiceProtocol,
+         keychainService: KeychainServiceProtocol,
+         coordinator: AuthCoordinatorProtocol,
+         analytics: AnalyticsManager) {
         self.authService = authService
         self.keychainService = keychainService
         self.coordinator = coordinator
+        self.analytics = analytics
     }
     
     // MARK: - Public
     
     func signIn() {
         guard !email.isEmpty && !password.isEmpty else { return }
         authService.signIn(withEmail: email, password: password) { [weak self] result in
-            self?.handleAuth(result: result)
+            switch result {
+            case .success(let user):
+                self?.user = user
+                
+                guard let currUser = Auth.auth().currentUser,
+                      let userData = currUser.uid.data(using: .utf8) else { return }
+                if let userEmail = currUser.email {
+                    self?.analytics.log(.signIn(email: userEmail))
+                }
+                self?.keychainService.set(userData, forKey: "user_id")
+                self?.coordinator.navigateToSettings()
+            case .failure(let error):
+                self?.authError = error
+            }
```

Comment:
Seems like duplicated code and you can use one function for it (57-70 lines). Just pass result and event type as a parameters.